docs/src/pages/theming.mdx

```diff
@@ -0,0 +1,430 @@
+  allowed content.
+</Callout>
+
+<Callout type="warning">
+  Note: UploadButton's button element is defined using `label`
+</Callout>
+
+import Image from "next/image";
+
+<Image
+  src="/upload-button-anatomy.png"
+  alt="Upload button anatomy"
+  width="1546"
+  height="998"
+  style={{ marginTop: "1rem", borderRadius: "0.5rem" }}
+/>
+
+### UploadDropzone
+
+Simplified component structure:
+
+```jsx
+<div className={className} data-state={getUtState()}>
+  <svg data-ut-element="upload-icon" data-state={getUtState()}>
+    ...
+  </svg>
+  <label data-ut-element="label" data-state={getUtState()}>
+    /label content goes here/
+    <input disabled={!ready} />
+  </label>
+  <div data-ut-element="allowed-content" data-state={getUtState()}>
+    /allowed content goes here/
+  </div>
+  <button
+    data-ut-element="button"
+    data-state={getUtState()}
+    disabled={isUploading}
+  >
+    /button content goes here/
+  </button>
+</div>
+```
+
+<Callout type="info">
+  UploadDropzone consists of five themable elements: container, upload icon,
+  label, button and allowed content.
+</Callout>
+
+<Callout type="warning">
+  Note: While in UploadButton the button element is defined using `label`, in
+  UploadDropzone it is defined using `button`. As an abstraction layer, button
+  element in these two components have special data attribute applied:
+  `data-ut-element="button"`.
+</Callout>
+
+<Image
+  src="/upload-dropzone-anatomy.png"
+  alt="Upload dropzone anatomy"
+  width="2010"
+  height="1594"
+  style={{ marginTop: "1rem", borderRadius: "0.5rem" }}
+/>
+
+## Theming props
+
+### `className`
+
+Both UploadButton and UploadDropzone accept `className` prop. It allows you to
+pass any additional classes to the component. All classes that are being passed
+through this prop are going to be applied to the outermost element - container
+
+### `appearance`
+
+Both `UploadButton` and `UploadDropzone` accept `appearance` prop. It accepts an
+object with keys that correspond to elements of component. You can pass
+following types of values:
+
+- `string` - it will be applied as a class to the element
+- `CSSProperties` object - it will be applied as an inline style to the element
+- `callback` - it will be called with current state of the component and should
+  return one of the above types
+
+The interfaces of `appearance` prop for `UploadButton` and `UploadDropzone` are
+following
+
+```ts
+type UploadButtonProps = {
+  /* rest of props */
+  appearance?: {
+    container?: StyleField;
+    button?: StyleField;
+    allowedContent?: StyleField;
+  };
+};
+
+type UploadDropzoneProps = {
+  /* rest of props */
+  appearance?: {
+    container?: StyleField;
+    uploadIcon?: StyleField;
+    label?: StyleField;
+    allowedContent?: StyleField;
+    button?: StyleField;
+  };
+};
+```
+
+The types of callback arguments are following:
+
+```ts
+type ButtonCallbackArguments = {
```

Comment:
Do we really need this callback? Is the data attributes not enough?

Reply:
While data attributes are enough when it comes to styling through classes, callback makes it possible to apply state-specific styles with inline styles